### PR TITLE
feat(nim-editor): add libnoise nim bindings:

### PR DESCRIPTION
**Pull request by Milerius**
- History done
- clear done
- exit cmd done

### CODE REVIEW

File: albinos-editor/src/albinospkg/cli.nim [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import terminal
+import rdstdin
+import linenoise
+
+func initCLI() {.inline.} =
+    discard historySetMaxLen(500)
+
+proc launchCLI*() =
+    initCLI()
+    while true:
```
Comment (Gaspard--) [inline albinos-editor/src/albinospkg/cli.nim]: Will this properly quit on ctrl-D ?

Comment (Milerius) [reply albinos-editor/src/albinospkg/cli.nim]: Ctrl-D is sigterm?

Comment (Gaspard--) [reply albinos-editor/src/albinospkg/cli.nim]: I think it's supposed to cause sigterm internally. IIRC in C at least you're supposed to check by looking if `read` returns 0.

File: albinos-editor/src/albinospkg/cli.nim [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+proc handleConfigCmd(argsSet: openArray[string]) =
+    echo argsSet
+
+proc launchCLI*() =
+    discard historySetMaxLen(500)
+    while true:
+        var line: TaintedString
+        let res = readLineFromStdin("> ", line)
+        if res == false:
+            styledEcho(fgMagenta, "readline error, quitting CLI mode...")
```
Comment (Gaspard--) [inline albinos-editor/src/albinospkg/cli.nim]:
```suggestion
            styledEcho(fgMagenta, "quitting CLI mode...")
```

**Pull request merged**